x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/BCrypt.java

```diff
@@ -652,7 +658,7 @@ private void ekskey(byte data[], byte key[]) {
      *                  using BCrypt.gensalt)
      * @return          the hashed password
      */
-    public static String hashpw(SecureString password, String salt) {
+    public static String generateHash(SecureString password, String salt) {
```

Comment:
@tvernum I missed the copyright notice. Agree on minimizing changes to it then. Besides not renaming any methods, it could be desirable to leave the original source as is, except for the specific check for new revisions:

`- if (minor != 'a' || salt.charAt(3) != '$')`
`+ if ((minor != 'a' && minor != 'b' && minor != 'y') || salt.charAt(3) != '$')`

This means that some supporting bcrypt code refactored from Hasher should return there or create a wrapper in order to better separate responsibilities while also maintaining the BCrypt class as close as the original as possible.

What do you think?